#165273579 : User can favorite article

nadralia opened the pull request:
#### What does this PR do?

Allow an authenticated user to favourite and unfavourite an article.
Prevent an unauthenticated user from favouriting and unfavouriting an article.
#### How should this be manually tested?
Make migrations and run migrate
`python manage.py makemigrations`
`python manage.py migrate`

Run the application:
`python manage.py runserver`
Open postman
Register a user using the route
http://127.0.0.1:8000/api/users/
Login the user you just registered
http://127.0.0.1:8000/api/users/login/
Get the token that was returned
Insert a header in postman with a key of Authorization and value Bearer <token>
Then visit http://127.0.0.1:8000/api/articles/ to create an article using POST.
Then visit  http://127.0.0.1:8000/api/articles/<slug>/favorite to either favourite  using POST or unfavourite an article DELETE.

#### What are the relevant pivotal tracker stories?

 [#165273579](https://www.pivotaltracker.com/story/show/165273579)

nadralia committed:
ft(favorite):User can favorite article

add favoritemodel
add post and delete favorite
write tests

[Finishes #165273579]

nadralia committed:
ft(favorite):User can favorite article

add favoritemodel
add post and delete favorite
write tests

[Finishes #165273579]

bisonlou commented on authors/apps/articles/tests/test_favorite_article.py:
```diff
@@ -0,0 +1,171 @@
+        self.token = self.user.token
+
+        # verify and save the second user
+        self.user2.is_verified = True
```
I do not know of this field in the user model.

nadralia replied:
@bisonlou Thanks for the feedback , this field is supposed to be 'is_email_verified'. let me fix this.

bisonlou commented on authors/apps/articles/views.py:
```diff
@@ -79,4 +90,72 @@ def put(cls, request, article_id):
+        if not favorited:
+            return Response(
+                {
+                    'message': "You have already unfavourited this article"
```
Since you were searching for a favorite article, on not finding it, the message should be "Article not found"

nadralia committed:
ft(favorite):User can favorite article

add favoritemodel
add post and delete favorite
write tests

[Finishes #165273579]
